docs/feature-requests/006-pull-request-reviewer-flow.md

```diff
@@ -0,0 +1,156 @@
+**_Part 1 - Required information_**
+
+# Pull Request Review -- Reviewer Flow
+
+## :memo: Summary
+
+Provide code review to an existing pull request within Atom.
+
+*Note*: This RFC is an iteration of the [original RFC for Pull Request Review](./003-pull-request-review.md).
+
+## :checkered_flag: Motivation
+
+We already have an innovative review-comments-in-dock (RCID) workflow built out for the receiving end of pull request reviews. In order to complete the full code review experience within Atom, we should also build out a workflow for users to author pull request reviews.
+
+## 🤯 Workflow Explanation
+
+This is a high level overview of what the workflow of a PR Review author should look like. More on the functionality and behaviour of each component in the next section.
+
+#### 1. Start a review
+
+There are three ways to start a review:
+1. Click "Start a review" button on the header of review dock, or footer of PR detail item, or on the empty state of review dock
+2. [Respond to an existing review thread by clicking "Start a review"](#responding-to-a-comment-thread)
+3. [Click on a "add comment" icon on the gutter](#add-comment-gutter-icon)
+
+#### 2. Continue a review
+
+Once a pending review has been started, user can add more comments to it by:
+1. [Responding to an existing review thread](#responding-to-a-comment-thread)
+2. [Clicking "add comment" icon on the gutter](#add-comment-gutter-icon)
+
+#### 3. Submit a review
+The only way to submit a review within Atom is by using the ["Submit review" button in the Pending Review tab](#summary-section). After publishing the review, the Pending Review tab will be destroyed. User will be led back to the All Reviews tab, which will immediately reflect the just published review.
+
+## 🤯 Components Explanation
+
+
+### "All Reviews" tab
+This tab shows all review summaries and review comments, including the ones that are part of a _pending review_ that has not been submitted yet.
+
+#### Header
+| current header | proposed header | proposed header with pending review |
+|---|---| --- |
+|<img width="380" alt="old header" src="https://user-images.githubusercontent.com/6842965/56446925-7f027600-62d3-11e9-8635-e17b946f3b1b.png">|<img width="380" alt="new header" src="https://user-images.githubusercontent.com/6842965/56446924-7f027600-62d3-11e9-98f5-0ea1e2826343.png">|<img width="380" alt="new header" src="https://user-images.githubusercontent.com/6842965/56447078-4911c180-62d4-11e9-8543-4935ba587b38.png">|
+
+- When there is no pending review, button reads "Start a review", clicking on which will take you to the Pending Review tab in its empty state.
```

Comment:
Can you start a review on a pull request that isn't checked out?

I wonder if we could keep the "Checkout" button when the pull request isn't checked out, then contextually turn it into either the "Start a review" or "Resume review" buttons when the pull request is checked out.

Reply:
:thinking: It probably makes sense to allow comments on PR without checking it out, just like you would on dotcom.